libs/distancing.py

```diff
@@ -80,29 +85,43 @@ def __process(self, cv_image):
         rgb_resized_image = cv.cvtColor(resized_image, cv.COLOR_BGR2RGB)
         tmp_objects_list = self.detector.inference(rgb_resized_image)
-        # Get the classifier result for detected face
-        if self.classifier is not None:
-            faces = []
-            for itm in tmp_objects_list:
-                if 'face' in itm.keys():
-                    face_bbox = itm['face']  # [ymin, xmin, ymax, xmax]
-                    if face_bbox is not None:
-                        xmin, xmax = np.multiply([face_bbox[1], face_bbox[3]], self.resolution[0])
-                        ymin, ymax = np.multiply([face_bbox[0], face_bbox[2]], self.resolution[1])
-                        croped_face = cv_image[
-                            int(ymin):int(ymin) + (int(ymax) - int(ymin)),
-                            int(xmin):int(xmin) + (int(xmax) - int(xmin))
-                        ]
-
-                        # Resizing input image
-                        croped_face = cv.resize(croped_face, tuple(self.classifier_img_size[:2]))
-                        croped_face = cv.cvtColor(croped_face, cv.COLOR_BGR2RGB)
-                        # Normalizing input image to [0.0-1.0]
-                        croped_face = np.array(croped_face) / 255.0
-                        faces.append(croped_face)
+
+        [w, h] = self.resolution
+        detection_scores = []
+        class_ids = []
+        detection_bboxes = []
+        faces = []
+        # Get the classifier of detected face
+        for itm in tmp_objects_list:
+            if 'face' in itm.keys() and self.classifier is not None:
+                face_bbox = itm['face']  # [ymin, xmin, ymax, xmax]
+                if face_bbox is not None:
+                    xmin, xmax = np.multiply([face_bbox[1], face_bbox[3]], self.resolution[0])
+                    ymin, ymax = np.multiply([face_bbox[0], face_bbox[2]], self.resolution[1])
+                    croped_face = cv_image[
+                        int(ymin):int(ymin) + (int(ymax) - int(ymin)),
+                        int(xmin):int(xmin) + (int(xmax) - int(xmin))
+                    ]
+
+                    # Resizing input image
+                    croped_face = cv.resize(croped_face, tuple(self.classifier_img_size[:2]))
+                    croped_face = cv.cvtColor(croped_face, cv.COLOR_BGR2RGB)
+                    # Normalizing input image to [0.0-1.0]
+                    croped_face = np.array(croped_face) / 255.0
+                    faces.append(croped_face)
+            # Prepare tracker input
+            box = itm["bbox"]
+            x0 = box[1]
+            y0 = box[0]
+            x1 = box[3]
+            y1 = box[2]
+            detection_scores.append(itm['score'])
+            class_ids.append(int(itm['id'].split('-')[0]))
+            detection_bboxes.append((int(x0 * w), int(y0 * h), int(x1 * w), int(y1 * h)))
             faces = np.array(faces)
             face_mask_results, scores = self.classifier.inference(faces)
```

Comment:
```suggestion
        face_mask_results, scores = self.classifier.inference(faces)
```

The faces and face_mask_results must be outside the `for`. Otherwise, after the first iteration the processor will throw the following exception:

```
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/repo/libs/engine_threading.py", line 46, in run
    self.engine.process_video(self.source['url'])
  File "/repo/libs/distancing.py", line 317, in process_video
    cv_image, objects, distancings = self.__process(cv_image)
  File "/repo/libs/distancing.py", line 111, in __process
    faces.append(croped_face)
AttributeError: 'numpy.ndarray' object has no attribute 'append'


```